samcli/commands/deploy/guided_context.py

```diff
@@ -285,52 +288,226 @@ def prompt_parameters(
+
         stacks : List[Stack]
             List of stacks to look for image functions.
 
+        image_repositories: Dict[str, str]
+            Dictionary with function logical ID as key and image repo URI as value.
+
+        region: str
+            Region for the image repos.
+
+        s3_bucket: str
+            s3 bucket URI to be used for uploading companion stack template
+
+        s3_prefix: str
+            s3 prefix to be used for uploading companion stack template
+
         Returns
         -------
-        Dict
+        Dict[str, str]
             A dictionary contains image function logical ID as key, image repository as value.
         """
-        image_repositories = {}
-        artifacts_format = get_template_artifacts_format(template_file=self.template_file)
-        if IMAGE in artifacts_format:
-            self.function_provider = SamFunctionProvider(stacks, ignore_code_extraction_warnings=True)
-            function_resources = get_template_function_resource_ids(template_file=self.template_file, artifact=IMAGE)
-            for resource_id in function_resources:
-                image_repositories[resource_id] = prompt(
-                    f"\t{self.start_bold}Image Repository for {resource_id}{self.end_bold}",
-                    default=self.image_repositories.get(resource_id, "")
-                    if isinstance(self.image_repositories, dict)
-                    else "" or self.image_repository,
-                )
-                if not is_ecr_url(image_repositories.get(resource_id)):
-                    raise GuidedDeployFailedError(
-                        f"Invalid Image Repository ECR URI: {image_repositories.get(resource_id)}"
-                    )
-            for resource_id, function_prop in self.function_provider.functions.items():
-                if function_prop.packagetype == IMAGE:
-                    image = function_prop.imageuri
-                    try:
-                        tag = tag_translation(image)
-                    except NonLocalImageException:
-                        pass
-                    except NoImageFoundException as ex:
-                        raise GuidedDeployFailedError("No images found to deploy, try running sam build") from ex
-                    else:
-                        click.secho(f"\t  {image} to be pushed to {image_repositories.get(resource_id)}:{tag}")
-            click.secho(nl=True)
+        image_repositories = image_repositories.copy() if image_repositories is not None else {}
+        self.function_provider = SamFunctionProvider(stacks, ignore_code_extraction_warnings=True)
+
+        manager_helper = CompanionStackManagerHelper(
+            stack_name, region, s3_bucket, s3_prefix, self.template_file, image_repositories
+        )
+
+        create_all_repos = self.prompt_create_all_repos(manager_helper)
+        if create_all_repos:
+            image_repositories.update(manager_helper.manager.get_repository_mapping())
```

Comment:
```
manager_helper.manager
```

Seems a bit off to me, abstraction wise. should it just be CompanionStack? or something else to showcase manager is contained within an entity that could contain it. Thoughts? I'm not sold on the naming that I proposed either.

Reply:
This is a helper class to bridge things that do not necessarily fit into CompanionStackManager but are needed for CLI purposes. I'm not sure about the naming of it either, CompanionStackManagerHelper just came to mind due to its purpose.